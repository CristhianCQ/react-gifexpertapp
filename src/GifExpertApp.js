import React, { useState } from 'react'
import {AddCategory} from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

const GifExpert= ({ defaultCategories = []}) =>{

    const [ categorie, setCategories]= useState(defaultCategories);

    // const  AddCategories= () => {
    //     // setCategories([...categorie,'Eden Zero']);
    //     setCategories(cats => [...cats, 'Eden Zero' ])
    // } 

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
            
            <ol>                
                {   
                    categorie.map((cat ) => 
                        <GifGrid 
                            key={cat}
                            category={cat}
                        />
                    )
                }               
            </ol>
        </>
    )
}

export default GifExpert;
 