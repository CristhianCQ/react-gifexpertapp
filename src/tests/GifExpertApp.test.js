
import React from 'react';
import '@testing-library/jest-dom';   
import { shallow } from 'enzyme';
import { default as GifExpert } from '../GifExpertApp';

describe('Pruebas en componente <GifExpertApp.js />',() => {

    test('debe de montarse el componente de manera adecuada',() => {

        const wrapper = shallow( <GifExpert />);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categorias = ['One Punch', 'Dragon ball'];

        const wrapper = shallow( <GifExpert  defaultCategories={categorias}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length); 

    })
    
})