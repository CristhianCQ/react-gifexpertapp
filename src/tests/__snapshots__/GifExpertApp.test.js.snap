// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifExpertApp.js /> debe de montarse el componente de manera adecuada 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente <GifExpertApp.js /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
